.github/workflows/ci.yaml

```diff
@@ -49,12 +49,10 @@ jobs:
 
   build:
     runs-on: ubuntu-22.04
-    needs: [lint,tests]
```

Comment:
I see you moved this from the `build` to the `upload` job. 
It means, you can still build untested assets even though they won't be deployed?

Is it possible to rather chain the premises so that, 
-  If `lint` and `tests` both succeed the assets are built
- If built is successful then trigger `upload` (add a `needs: build` on `upload`) ?

Beside sparing some action time, this could also be useful so potentially broken builds are not stored and ditched when re-running workflows